README.Rmd

```diff
@@ -49,136 +48,85 @@ pak::pak("tidyverse/glue")
+glue("My name is {name}.")
 ```
 
-`glue::glue()` is also made available via `stringr::str_glue()`.
-So if you've already attached stringr (or perhaps the whole tidyverse), you can access `glue()` like so:
+`stringr::str_glue()` is an alias for `glue::glue()`.
+So if you've already attached stringr (or perhaps the whole tidyverse), you can use `str_glue()` to access all of the functionality of `glue()`:
 
 ```{r eval = FALSE}
 library(stringr) # or library(tidyverse)
 
-stringr_fcn <- "`stringr::str_glue()`"
-glue_fcn    <- "`glue::glue()`"
-
-str_glue('{stringr_fcn} is essentially an alias for {glue_fcn}.')
-#> `stringr::str_glue()` is essentially an alias for `glue::glue()`.
+name <- "Wilma"
+str_glue("My name is {name}.")
+#> My name is Wilma.
 ```
 
-`glue_data()` works well with pipes:
+You're not limited to using a bare symbol inside `{}`; it can basically be any little bit of R code:
 
 ```{r}
-mtcars$model <- rownames(mtcars)
-mtcars |> head() |> glue_data("{model} has {hp} hp")
+name <- "Pebbles"
+glue("Here is my name in uppercase and doubled: {strrep(toupper(name), 2)}.")
 ```
 
-##### `glue_data()` is useful with [magrittr](https://cran.r-project.org/package=magrittr) pipes.
-```{r}
-`%>%` <- magrittr::`%>%`
-head(mtcars) %>% glue_data("{rownames(.)} has {hp} hp")
-```
+### Data can come from various sources
 
-##### `glue()` is useful within dplyr pipelines
-```{r, message = FALSE}
-library(dplyr)
-head(iris) %>%
-  mutate(description = glue("This {Species} has a petal length of {Petal.Length}"))
-```
+glue can interpolate values from the local environment or from data passed in `name = value` form:
 
-##### Leading whitespace and blank lines from the first and last lines are automatically trimmed.
-This lets you indent the strings naturally in code.
 ```{r}
-glue("
-    A formatted string
-    Can have multiple lines
-      with additional indention preserved
-    ")
+x <- "the local environment"
+glue(
+  "`glue()` can access values from {x} or from {y}. {z}",
+  y = "named arguments",
+  z = "Woo!"
+)
 ```
 
-##### An additional newline can be used if you want a leading or trailing newline.
-```{r}
-glue("
-
-  leading or trailing newlines can be added explicitly
+If the relevant data lives in a data frame (or list or environment), use `glue_data()` instead:
 
-  ")
-```
-
-##### `\\` at the end of a line continues it without a new line.
 ```{r}
-glue("
-    A formatted string \\
-    can also be on a \\
-    single line
-    ")
+mini_mtcars <- head(cbind(model = rownames(mtcars), mtcars))
+glue_data(mini_mtcars, "{model} has {hp} hp.")
 ```
 
-##### A literal brace is inserted by using doubled braces.
-```{r}
-name <- "Fred"
-glue("My name is {name}, not {{name}}.")
-```
+`glue_data()` is very natural to use with the pipe:
 
-##### Alternative delimiters can be specified with `.open` and `.close`.
 ```{r}
-one <- "1"
-glue("The value of $e^{2\\pi i}$ is $<<one>>$.", .open = "<<", .close = ">>")
+mini_mtcars |>
```

Comment:
I'm crossing the Rubicon of showing the base pipe in glue's documentation. Doesn't really matter in `README.Rmd`, but it's the beginning of the larger transition.